# Task 33 — Customers by Subscription Fee Range

## Business Question

How many customers have a monthly subscription fee between $100 and $200?

## Analysis

Filtered customers whose monthly_fee is between 100 and 200 (inclusive), then grouped the results by subscription plan and counted the number of customers in each plan.

## Code

In [1]:
import pandas as pd
import numpy as np
customers = pd.read_csv("starlink_customers.csv")
subscriptions = pd.read_csv("subscriptions.csv")
starlink = customers.merge(
    subscriptions,
    left_on = "unique_id",
    right_on = "customer_id",
    how = "left"
)
starlink[starlink["monthly_fee"].between(100,200)].groupby("plan_name").agg(
    users_amount = ("unique_id","count")
)

,users_amount
plan_name,
Residential,5959


### Business Insight

All customers within the $100–$200 monthly fee range are subscribed to the Residential plan. The Residential plan has a monthly fee of $120 and accounts for all 5,959 customers in this price range.